compiler/rustc_typeck/src/coherence/mod.rs

```diff
@@ -75,6 +75,19 @@ fn enforce_trait_manually_implementable(
         return;
     }
 
+    if did == li.as_repr_trait() {
+        let span = impl_header_span(tcx, impl_def_id);
+        struct_span_err!(
+            tcx.sess,
+            span,
+            E0322,
+            "explicit impls for the `AsRepr` trait are not permitted"
+        )
+        .span_label(span, "impl of 'AsRepr' not allowed")
```

Comment:
Can we also mention in a note or as part of the label or suggestion that they should `derive` instead?

Reply:
Done - I'll also take a look at porting the `FromRepr` diagnostics over so that we can suggest something like "If you added a repr it would be automatically implemented" or "This data-holding variant is the thing preventing an auto implementation", but may leave that for a follow-up if that works for you